Automattic/wp-calypso
# Help Center: localize queries

This PR localizes the queries we use for searching help center articles.
Based on the route where the customer is, we use the appropriate query to show relevant articles when the help center is opened.

For example in italian locale if we open the help center when we are on the plans page we see empty suggestions:
<img width="1918" alt="Screenshot 2024-08-21 at 19 54 01" src="https://github.com/user-attachments/assets/29420c6e-1bb1-42f2-9f3e-f8207c793ce6">
This is because we are searching through the italian support page with query `upgrade plan`.
In this case the search should be with translated query `piano di aggiornamento`. 
It should bring up result is if the customer searches this query in his locale:
<img width="897" alt="Screenshot 2024-08-21 at 19 54 33" src="https://github.com/user-attachments/assets/d4771e17-bc63-45d7-9f93-a5905b2d52ce">

Fixed https://github.com/Automattic/dotcom-forge/issues/8844

## Proposed Changes

* Translate queries mapped to routes

## Why are these changes being made?


* In locales other than english we were using english queries to search

## Testing Instructions
- Use PR link
- Switch to locale other than english
- Navigate to routes that are targeted in the `route-to-query-mapping`
- Verify that translated query is used when searching for articles




## Pre-merge Checklist
